blob/driver/driver.go

```diff
@@ -65,6 +65,9 @@ type WriterOptions struct {
 	// write in a single request, if supported. Larger objects will be split into
 	// multiple requests.
 	BufferSize int
+	// Content-MD5 which may be used as a message integrity check (MIC)
```

Comment:
Hi @myml, thanks for the contribution!

`blob` and `blob/driver` are both in the same module, so you shouldn't need to split this change up into multiple Pull Requests.

Also, I'd like to see it working, including the implementation for `s3blob` and `gcsblob` (these should be easy, just pass-through to the provider) and `fileblob` (might be trickier...), and a test in `drivertest/drivertest.go`.